### PR TITLE
Issue #4193: The `file_temporary_path` is not initially set on `system.core.json` file.

### LINKED ISSUE #4193: Set `file_temporary_path` in initial `system.core.json` file
**Description of the bug**

Despite the `/admin/config/media/file-system` page on a freshly installed Backdrop site shows that **Temporary directory** is set to `/tmp`:

Screen Shot 2019-11-03 at 3 46 43 AM

the `file_temporary_path` variable in the `system.core.json` file in active configuration directory shows empty. Therefore the function `config_get('system.core', 'file_temporary_path')` returns empty value causing number of serious troubles.

**Steps To Reproduce**

Just deploy a new Backdrop site and see the described difference between UI and configuration file.

You can quickly fire up the `cat files/config_*/active/system.core.json | grep file_temporary_path` command on CLI to see:

```
    "file_temporary_path": "",
```

At the same time, if you go to the `/admin/config/media/file-system` page then it shows `/tmp` for Temporary directory.

**Expected behavior**

The `file_temporary_path` variable in `system.core.json` file should match what the website's user interface shows for **Temporary directory**.

**Additional information**

If you save the `/admin/config/media/file-system` page, then the `file_temporary_path` variable in `system.core.json` file gets correct value. However, we should not expect that every user necessarily goes to and saves the `/admin/config/media/file-system` page just after installation. 

To me this seems pretty urgent issue to fix, so I'll be submitting PR soon.

---
~~PR by @alanmels: https://github.com/backdrop/backdrop/pull/2971~~
~~New PR by @alanmels: https://github.com/backdrop/backdrop/pull/2972~~
Yet another PR by @alanmels: https://github.com/backdrop/backdrop/pull/2988

### PR DESCRIPTION
**Pull request by alanmels**
Fixes https://github.com/backdrop/backdrop-issues/issues/4193

### CODE REVIEW

File: core/modules/system/system.install [hunk 1]
```diff
@@ -245,9 +245,11 @@ function system_requirements($phase) {
         'directory' => $private_files_directory,
       );
     }
+    $temporary_directory = file_directory_temp();
+    config_set('system.core', 'file_temporary_path', $temporary_directory);
     $htaccess_files['temporary://.htaccess'] = array(
       'title' => $t('Temporary files directory'),
-      'directory' => file_directory_temp(),
+      'directory' => $temporary_directory,
```
Comment (indigoxela) [inline core/modules/system/system.install:252]: This has at least one drawback: it only strikes if backdrop_is_apache().

Eventually it's better to do this in phase 'install' and/or 'update'. Not sure about that yet.

Comment (alanmels) [reply core/modules/system/system.install:252]: Please also read https://github.com/backdrop/backdrop-issues/issues/4193#issuecomment-550338846 

So we are back to my first PR?

Comment (indigoxela) [reply core/modules/system/system.install:252]: I'm not the one to decide in any way, I just comment possible glitches here. ;-)
Sorry, I'm currently too busy to do more.

**Comment by alanmels** [comment]
Closed per https://github.com/backdrop/backdrop-issues/issues/4193#issuecomment-552068391

**Pull request closed without merging**